cognite_toolkit/cdf_tk/load.py

```diff
@@ -403,16 +406,42 @@ def get_required_capability(cls, ToolGlobals: CDFToolConfig) -> Capability:
     def get_id(cls, item: Group) -> str:
         return item.name
 
-    def load_file(self, filepath: Path, ToolGlobals: CDFToolConfig) -> Group:
+    def replace_scope_ids(self, capability: dict, ToolGlobals: CDFToolConfig):
+        if len(capability.get("scope", {}).get("datasetScope", {}).get("ids", [])) > 0:
+            capability["scope"]["datasetScope"]["ids"] = [
+                ToolGlobals.verify_dataset(ext_id)
+                for ext_id in capability.get("scope", {}).get("datasetScope", {}).get("ids", [])
+            ]
+
+        if len(capability.get("scope", {}).get("extractionPipelineScope", {}).get("ids", [])) > 0:
+            capability["scope"]["extractionPipelineScope"]["ids"] = [
+                ToolGlobals.verify_extraction_pipeline(ext_id)
+                for ext_id in capability.get("scope", {}).get("extractionPipelineScope", {}).get("ids", [])
+            ]
+        return capability
+
+    def load_file(self, filepath: Path, ToolGlobals: CDFToolConfig) -> Group | None:
         raw = load_yaml_inject_variables(filepath, ToolGlobals.environment_variables())
+
+        resource_scope_str = [cls.__name__.lower() for cls in self.resource_scopes]
+
+        to_create = []
+
         for capability in raw.get("capabilities", []):
-            for _, values in capability.items():
-                if len(values.get("scope", {}).get("datasetScope", {}).get("ids", [])) > 0:
-                    values["scope"]["datasetScope"]["ids"] = [
-                        ToolGlobals.verify_dataset(ext_id)
-                        for ext_id in values.get("scope", {}).get("datasetScope", {}).get("ids", [])
-                    ]
-        return Group.load(raw)
+            for acl, values in capability.items():
```

Comment:
Not sure what this section is relative to what I have done in #180 , but this can become really messy to merge. But maybe the pattern in #180 can be reused?